h5py/_hl/vds.py

```diff
@@ -51,9 +51,21 @@ class VirtualSource(object):
         The source dataset is resizable up to this shape. Use None for
         axes you want to be unlimited.
     """
-    def __init__(self, path_or_dataset, name=None, shape=None, dtype=None, maxshape=None):
+    def __init__(self, path_or_dataset, name=None,
+                 shape=None, dtype=None, maxshape=None):
         from .dataset import Dataset
         if isinstance(path_or_dataset, Dataset):
+            if any(inp is not None
+                   for inp in [name, shape, dtype, maxshape]):
+                failed = {k: v
+                          for k, v in
+                          {'name': name, 'shape': shape,
+                           'dtype': dtype, 'maxshape': maxshape}.items()
```

Comment:
Slightly more concise: construct this dict before the `if`, then just check `if failed:`.